contrib/rtlbrowse/logfile.c

```diff
@@ -93,1390 +89,1308 @@ static GtkTextTag *blue_tag = NULL, *fwht_tag = NULL;
 {
-GtkTextBuffer *tb = gtk_text_view_get_buffer(GTK_TEXT_VIEW(tr->text));
-GtkTextMark *tm = gtk_text_buffer_get_insert(tb);
-GtkTextIter iter;
-gint llen;
-
-gtk_text_buffer_get_iter_at_mark(tb, &iter, tm);
-gtk_text_iter_set_line(&iter, tr->line);
-llen = gtk_text_iter_get_chars_in_line (&iter);
-tr->offs = (tr->offs > llen) ? llen : tr->offs;
-gtk_text_iter_set_line_offset(&iter, tr->offs);
-
-gtk_text_buffer_place_cursor(tb, &iter);
+    GtkTextBuffer *tb = gtk_text_view_get_buffer(GTK_TEXT_VIEW(tr->text));
+    GtkTextMark *tm = gtk_text_buffer_get_insert(tb);
+    GtkTextIter iter;
+    gint llen;
+
+    gtk_text_buffer_get_iter_at_mark(tb, &iter, tm);
+    gtk_text_iter_set_line(&iter, tr->line);
+    llen = gtk_text_iter_get_chars_in_line(&iter);
+    tr->offs = (tr->offs > llen) ? llen : tr->offs;
+    gtk_text_iter_set_line_offset(&iter, tr->offs);
+
+    gtk_text_buffer_place_cursor(tb, &iter);
 }
 
-static void
-forward_chars_with_skipping (GtkTextIter *iter,
-                             gint         count)
+static void forward_chars_with_skipping(GtkTextIter *iter, gint count)
 {
+    gint i;
 
-  gint i;
+    g_return_if_fail(count >= 0);
 
-  g_return_if_fail (count >= 0);
+    i = count;
 
-  i = count;
+    while (i > 0) {
+        gboolean ignored = FALSE;
 
-  while (i > 0)
-    {
-      gboolean ignored = FALSE;
-
-      if (gtk_text_iter_get_char (iter) == 0xFFFC)
-        ignored = TRUE;
+        if (gtk_text_iter_get_char(iter) == 0xFFFC)
+            ignored = TRUE;
 
-      gtk_text_iter_forward_char (iter);
+        gtk_text_iter_forward_char(iter);
 
-      if (!ignored)
-        --i;
+        if (!ignored)
+            --i;
     }
 }
 
-static gboolean iter_forward_search_caseins(
-                              const GtkTextIter *iter,
-                              gchar             *str,
-                              GtkTextIter       *match_start,
-                              GtkTextIter       *match_end)
+static gboolean iter_forward_search_caseins(const GtkTextIter *iter,
+                                            gchar *str,
+                                            GtkTextIter *match_start,
+                                            GtkTextIter *match_end)
 {
-GtkTextIter start = *iter;
-GtkTextIter next;
-gchar *line_text, *found, *pnt;
-gint offset;
-gchar *strcaseins;
-
-if(!str) return(FALSE);
-
-pnt = strcaseins = strdup(str);
-while(*pnt)
-	{
-	*pnt = toupper((int)(unsigned char)*pnt);
-	pnt++;
-	}
-
-for(;;)
-	{
-	next = start;
-	gtk_text_iter_forward_line (&next);
-
-	/* No more text in buffer */
-	if (gtk_text_iter_equal (&start, &next))
-		{
-		free(strcaseins);
-	      	return(FALSE);
-	    	}
+    GtkTextIter start = *iter;
+    GtkTextIter next;
+    gchar *line_text, *found, *pnt;
+    gint offset;
+    gchar *strcaseins;
+
+    if (!str)
+        return (FALSE);
+
+    pnt = strcaseins = strdup(str);
+    while (*pnt) {
+        *pnt = toupper((int)(unsigned char)*pnt);
```

Comment:
This is outside the scope of this PR, but in the future we need to decide how text encodings will be handled by rtlbrowse and GTKWave in general. This function assumes that the input is UTF-8 encoded below by using `g_utf8_strlen`, but also uses `toupper` to convert to uppercase, which might or might not use UTF-8.

To fix this properly we need to get rid of the `WAVE_LOCALE_FIX`, which already causes encoding issues when `rtlbrowse --help` tries to displays some localized german help text with umlauts.